test/e2e/toolchaincluster_test.go

```diff
@@ -24,17 +24,36 @@ func TestToolchainClusterE2E(t *testing.T) {
 	memberAwait := awaitilities.Member1()
 	memberAwait.WaitForToolchainClusterResources(t)
 
-	verifyToolchainCluster(t, hostAwait.Awaitility, memberAwait.Awaitility)
-	verifyToolchainCluster(t, memberAwait.Awaitility, hostAwait.Awaitility)
+	verifyToolchainCluster(t, hostAwait.Awaitility, memberAwait.Awaitility, true)
+
+	// NOTE: The labels are currently only set in the host operator. In the next upgrade step, the member operator will
+	// also be updated to use the new version of the toolchain cluster controller.
+	verifyToolchainCluster(t, memberAwait.Awaitility, hostAwait.Awaitility, false)
 }
 
 // verifyToolchainCluster verifies existence and correct conditions of ToolchainCluster CRD
 // in the target cluster type operator
-func verifyToolchainCluster(t *testing.T, await *wait.Awaitility, otherAwait *wait.Awaitility) {
+func verifyToolchainCluster(t *testing.T, await *wait.Awaitility, otherAwait *wait.Awaitility, shouldHaveReferencedSecret bool) {
 	// given
 	current, ok, err := await.GetToolchainCluster(t, otherAwait.Namespace, nil)
 	require.NoError(t, err)
 	require.True(t, ok, "ToolchainCluster should exist")
+
+	// NOTE: this needs to run first, before the sub-tests below, because they reuse the secret for the new
+	// toolchain clusters. This is technically incorrect but sufficient for those tests.
+	// Note that we are going to be changing the workflow such that the label on the secret will actually be the driver
+	// for ToolchainCluster creation and so re-using the secret for different TCs will become impossible in the future.
```

Comment:
I don't quite understand. If reusing the secret for different TCs will be impossible in the future then are the other tests still going to be valid? If they're going to be invalid then we can just remove them now?

Reply:
Ah, is it supposed to mean the subtests will be updated in a following PR to create the secret with label instead of the ToolchainCluster directly?

Reply:
yep. I didn't want to rewrite the tests just yet, because they are still working.